pkg/pkgmgr/apk.go

```diff
@@ -215,9 +215,31 @@ func (am *apkManager) upgradePackages(ctx context.Context, updates unversioned.U
 		}
 	}
 
+	// If the image has been patched before, diff the base image and patched image to retain previous patches
+	if am.config.PatchedConfigData != nil {
+		// Diff the base image and new patches
+		patchDiff := llb.Diff(am.config.ImageState, apkInstalled)
+
+		// Diff the base image and patched image to get previous patches
+		prevPatchDiff := llb.Diff(am.config.ImageState, am.config.PatchedImageState)
+
+		// Merge the old patch diffs and new patch diffs
+		prevAndNewPatch := llb.Merge([]llb.State{prevPatchDiff, patchDiff})
+
+		// This additional diff ensures previous patch layers are discarded
+		// While ensuring all previous and new patches are properly applied to the base image
+		combinedPatch := llb.Diff(am.config.ImageState, prevAndNewPatch)
```

Comment:
It looks like there is a bug somewhere here.

When running copa patch with a trivy scan, and then running copa patch with update all, the resulting image cannot be run, and it looks like the file system is discarded inspecting the image with dive. The filesystem should be retained with only the new changes reflected in /usr/var/lib

Reply:
(same for the other package managers)